Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.h

```diff
@@ -269,6 +263,22 @@ namespace Terrain
             AZStd::shared_ptr<ProcessAsyncParams> params = nullptr) const override;
 
     private:
+        //! Given a set of async parameters, calculate the max number of jobs that we can use for the async call.
+        int32_t CalculateMaxJobs(AZStd::shared_ptr<ProcessAsyncParams> params) const
```

Comment:
nit, I know it's small but since it's not templated it could be in the cpp. It'd only be like 1 total extra line of code. Def not a blocking issue though.